### ISSUE TITLE
MacOS - Intel, fatal error: 'inc_vendor.h' file not found

### ISSUE THREAD

**Issue by eli-rich** [hashcat/hashcat#3080]
**Describe the bug**
I get this error when running hash cat on my MacBook Pro running macOS 11.6 Big Sur. It is an intel Mac.
I saw many others had similar issues but the solutions did not seem to work for me.

**To Reproduce**
`git clone https://github.com/hashcat/hashcat.git`
`cd hashcat`
`make install` (note: I get the exact same result with `make`)
`./hashcat -b`

**Hardware/Compute device (please complete the following information):**
- Compute device name: Intel(R) Core(TM) i5-1038NG7
- "OpenCL API (OpenCL 1.2 (Aug 30 2021 03:51:40)) - Platform 1 [Apple]"

**Hashcat version (please complete the following information):**
 - OS: macOS 11.6 Big Sur
 - Version: v6.2.5-55-gcb7f99ef7

**Error Output**
```
./hashcat -b
```
```
hashcat (v6.2.5-55-gcb7f99ef7) starting in benchmark mode

Benchmarking uses hand-optimized kernel code by default.
You can use it in your cracking session by setting the -O option.
Note: Using optimized kernel code limits the maximum supported password length.
To disable the optimized kernel code in benchmark mode, use the -w option.

OpenCL API (OpenCL 1.2 (Aug 30 2021 03:51:40)) - Platform #1 [Apple]
====================================================================
* Device #1: Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz, 8160/16384 MB (2048 MB allocatable), 8MCU
* Device #2: Intel(R) Iris(TM) Plus Graphics, skipped

Benchmark relevant options:
===========================
* --optimized-kernel-enable

-------------------
* Hash-Mode 0 (MD5)
-------------------

clCompileProgram(): CL_BUILD_PROGRAM_FAILURE

<program source>:7:10: fatal error: 'inc_vendor.h' file not found
#include "inc_vendor.h"
         ^

* Device #1: Kernel /Users/elirichardson/Documents/hashcat/OpenCL/shared.cl build failed.

Started: Sat Dec 25 01:46:56 2021
Stopped: Sat Dec 25 01:46:56 2021
```

**Full Output in File**
[log.txt](https://github.com/hashcat/hashcat/files/7775617/log.txt)

My sincerest apologies if I have simply overlooked the solution.

**Comment by pich4ya** [comment]
I can confirm that the same error occurs to me.

**Comment by matrix** [comment]
@BlazingFire007 please provide the full output for each commands you use

**Comment by eli-rich** [comment]
@matrix The full output should be in the log.txt file. If I've accidentally left anything out please let me know and I can try again!

**Comment by matrix** [comment]
@BlazingFire007 retry without "make install", so only "make" and then "./hashcat -b -m 0"

**Comment by eli-rich** [comment]
Here's the full log with only the make command
[log.txt](https://github.com/hashcat/hashcat/files/7791961/log.txt)
.

**Comment by matrix** [comment]
> Here's the full log with only the make command [log.txt](https://github.com/hashcat/hashcat/files/7791961/log.txt) .

why "sudo make"? 
Please attach the output of "hashcat -I" and clinfo 
(it would be better to discuss this on discord if you agree)

**Comment by eli-rich** [comment]
Update:
Using `sudo su` and running `make` as well as `hashcat` as root fixes the problem. I'm not sure why.

**Comment by eli-rich** [comment]
Solution:

I have no clue why this worked, nor do I know how this worked. But simply cloning the folder into my home directory and installing in `~/hashcat` instead of `~/Documents/hashcat` fixed the issue for me. Even works as user!

**Comment by JulesLePrince** [comment]
> @BlazingFire007 retry without "make install", so only "make" and then "./hashcat -b -m 0"

I had the same error :

```
clCompileProgram(): CL_BUILD_PROGRAM_FAILURE

<program source>:7:10: fatal error: 'inc_vendor.h' file not found
#include "inc_vendor.h"
         ^

* Device #1: Kernel /Users/elirichardson/Documents/hashcat/OpenCL/shared.cl build failed.
```

I solved it by calling ./hashcat directly in the makefile directory. Do you know how I could get the bash alias working?

**Comment by pich4ya** [comment]
> Solution:
> 
> I have no clue why this worked, nor do I know how this worked. But simply cloning the folder into my home directory and installing in `~/hashcat` instead of `~/Documents/hashcat` fixed the issue for me. Even works as user!

This surprisedly works.

**Comment by bbirkinbine** [comment]
Not sure why this works for me.

Intel (not M1), not running as root, MacOS BigSur, installed via brew install hashcat

If I call the hashcat command from the /usr/local/Cellar/hashcat/6.2.5/share/hashcat/OpenCL directory (the location of the inc_vendor.h), I no longer get the inc_vendor.h error.

**Comment by cbowns** [comment]
Figured this out: had to set `CPLUS_INCLUDE_PATH=/usr/local/share/hashcat/OpenCL`; i.e. in zsh I ran:
```
export CPLUS_INCLUDE_PATH=/usr/local/share/hashcat/OpenCL
```

and then I could run `hashcat` from wherever.

**Comment by daynaskully** [comment]
> Not sure why this works for me.
> 
> Intel (not M1), not running as root, MacOS BigSur, installed via brew install hashcat
> 
> If I call the hashcat command from the /usr/local/Cellar/hashcat/6.2.5/share/hashcat/OpenCL directory (the location of the inc_vendor.h), I no longer get the inc_vendor.h error.

Calling from the directory worked for me in the short term

- apple M1 pro
- osx 12.2.1
- zsh 5.8
- homebrew 3.3.16
- hashcat 6.2.5

**Comment by matrix** [comment]
> > Not sure why this works for me.
> > Intel (not M1), not running as root, MacOS BigSur, installed via brew install hashcat
> > If I call the hashcat command from the /usr/local/Cellar/hashcat/6.2.5/share/hashcat/OpenCL directory (the location of the inc_vendor.h), I no longer get the inc_vendor.h error.
> 
> Calling from the directory worked for me in the short term
> 
> * apple M1 pro
> * osx 12.2.1
> * zsh 5.8
> * homebrew 3.3.16
> * hashcat 6.2.5

Homebrew version is not updated with the latest changes. Please recompile the sources from github and test it again

**Comment by thomashooo** [comment]
> Solution:
> 
> I have no clue why this worked, nor do I know how this worked. But simply cloning the folder into my home directory and installing in `~/hashcat` instead of `~/Documents/hashcat` fixed the issue for me. Even works as user!

YES!!!!

**Comment by EricFromCanada** [comment]
The [Homebrew formula](https://formulae.brew.sh/formula/hashcat) installs the most recent release, and quite a few Mac-related fixes have been added since then (including one for [this issue](https://github.com/hashcat/hashcat/pull/3140)). The latest sources can be compiled and installed with `brew install --HEAD hashcat`, although running [this command](https://github.com/hashcat/hashcat/issues/3044#issuecomment-1116221725) to work around #3044 is still required (for now).

**Comment by D4DB0DY** [comment]
> The [Homebrew formula](https://formulae.brew.sh/formula/hashcat) installs the most recent release, and quite a few Mac-related fixes have been added since then (including one for [this issue](https://github.com/hashcat/hashcat/pull/3140)). The latest sources can be compiled and installed with `brew install --HEAD hashcat`, although running [this command](https://github.com/hashcat/hashcat/issues/3044#issuecomment-1116221725) to work around #3044 is still required (for now).

I am getting this... 

Error: Failure while executing; `patch -g 0 -f -p1` exited with 1. Here's the output:
patching file 'src/filehandling.c'
1 out of 1 hunks failed--saving rejects to 'src/filehandling.c.rej'

**Comment by EricFromCanada** [comment]
A missing include was just patched in #3387, so commenting out the `patch :DATA` line in your local copy of the formula file should get the HEAD build working again.

**Comment by afreeland** [comment]
Just dropping this in for anyone else that was unsure how to comment out the `patch :DATA` portion in local formula, the file can be found `/opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/hashcat.rb`
